Fixing and adding missing abstracts in the Logging guide (#35022)
quarkusio/quarkus: docs/src/main/asciidoc/logging.adoc

[[jboss-logging]]
== Use JBoss Logging for application logging

> Using the JBoss Logging API, your application requires no additional dependencies, as Quarkus automatically provides it.

Reviewer: ```suggestion
When using the JBoss Logging API, your application requires no additional dependencies, as Quarkus automatically provides it.
```

Otherwise it doesn't look like a sentence to me?

== Get an application logger

> In Quarkus, the most common ways to obtain an application logger are by <<declaring-a-logger-field,declaring a logger field>>, <<simplified-logging,simplified logging>>, and <<injection-of-a-configured-logger,injecting a configured logger>>.

Reviewer 2: To minimize future anchor maintenance (uses the upstream anchor post-fix script) and to support doc skimming: 

```suggestion
In Quarkus, the most common ways to obtain an application logger are by:

* <<declaring-a-logger-field>>
* <<simplified-logging>>
* <<injection-of-a-configured-logger>>
```

Reviewer 3: @michelle-purcell can you explain the issue with the anchors? Does that mean downstream can take "text" when using the << link >> syntax? In this case what text does it use?

Author: @cescoffier Hello!
 the `<<declaring-a-logger-field,declaring a logger field>>` approach I am using in the sentence is because I dont want to have Upper capitalization when listing the methods. @michelle-purcell suggests me to return to my previous version in which I had a bullet list containing these approaches. Since they are in a list, they can be naked and their Heading original names will be sourced automatically. 
 
 Yes, downstream limitations doesn't render naked link, thus the name needs to be specified always. However, @gsmet wrote a script that solve this issue, thus Quarkus contributors can use anytime.

[[injection-of-a-configured-logger]]
=== Injecting a configured logger

> The injection of a configured `org.jboss.logging.Logger` logger instance with the `@Inject` annotation is another alternative of adding an application logger, but applicable only to CDI beans.

Reviewer 2: ```suggestion
The injection of a configured `org.jboss.logging.Logger` logger instance with the `@Inject` annotation is another alternative to adding an application logger, but is applicable only to CDI beans.
```

== Logging format

Quarkus uses a pattern-based logging formatter that generates human-readable text logs by default.
Quarkus uses a pattern-based logging formatter that generates human-readable text logs by default, but you can also  configure the format for each log handler by using a dedicated property.

> You can configure the format for each log handler by using a dedicated property.

Reviewer: Not exactly a fan of making the abstract overly long...

=== File log handler

> The Quarkus file log handler is disabled by default, but when enabled, it logs events to a file on the host, supporting log file rotation.

Reviewer 2: Or, you could also describe it from the user's perspective/goal:

```suggestion
To log events to a file on the host, use the Quarkus file log handler, which is disabled by default.

The Quarkus file log handler supports log file rotation. 
Log file rotation ensures effective log file management over time by maintaining a specified number of backup log files, while also keeping the primary log file up-to-date and manageable.
```

Side note: :grey_question:  Wondering why this Quarkus feature wasn't called _Log file handler_.

By default, the file log handler in Quarkus is disabled.
Once enabled, it enables the logging of all events to a file on the application's host, while also supporting log file rotation.

> 

Reviewer: The previous sentence ends with mentioning log file rotation, so why do we start a new paragraph here?

We could alternatively move the first mention of log rotation to the next paragraph.

Author: Abstracts are limited to 26 words. Empty lines improve readability by dividing paragraphs into smaller chunks suitable for the reference style.

However, level 3 headings (=== XXX) don't require an abstract since they are subsections of a chapter with an abstract, so here it was just about that splitting.

What I will do is to patch this section so that it begins in more literal way :) Thank you for pointing this out.

Author: My applied tweak:

```
The Quarkus file log handler is disabled by default, but when enabled, it logs events to a file on the host, supporting log file rotation.

Log file rotation ensures effective log file management over time by maintaining a specified number of backup log files, while keeping the primary log file up-to-date and manageable.
```
@Ladicek If OK in your books, please, close the thread. :)

Reviewer: I think the previous version was better. Since we're not creating an abstract here, we don't need to condense as much information as possible into a single sentence.

Here's what I'd use:

> By default, the file log handler in Quarkus is disabled.
> Once enabled, it writes log messages to a file on the application's host.
>
> The file log handler also supports log file rotation.
> Log file rotation ensures effective log file management over time by maintaining a specified number of backup log files, while keeping the primary log file up-to-date and manageable.

=== File log handler

> To log events to a file on the application's host, use the Quarkus file log handler, which is disabled by default.

Reviewer: ```suggestion
To log events to a file on the application's host, use the Quarkus file log handler.
The file log handler is disabled by default, so you must first enable it.
```

== Logging configurations examples

> Frequently used logging configurations examples.

Reviewer: Not a sentence at all?

Author: What about:
Obtain the proper logger settings from the frequently used logging configurations examples.

Reviewer: Frankly, there's no need for an abstract here. The heading says it perfectly. But if we need something, how about this?

> Following are examples of frequently used logging configuration.

I know we don't like "follow" or "in this chapter" or so, but how exactly can we describe a collection of examples? :-)

Author: Thats another bad influence of playing Diablo 4 with all these typos and misconceptions :)

Author: I used:

```
Obtain the proper logger settings from the frequently used logging configurations examples.
```
Can imagine this line being used as a section description TBH.
Close the thread if OK.

@michelle-purcell , or do you have any better idea how to solve this? :)

Reviewer: I'd say we can obtain inspiration from examples, but often not "proper settings" :-) Would be good to have something better, for sure.

Author: ```
Get inspiration for real-world logger settings from frequently used logging configuration examples.
```
:)

== Logging configurations examples

> Get inspiration for real-world logger settings from examples of frequently used logging configurations.

Reviewer 2: Sorry your comment got buried in my view @MichalMaler.

How about...
```suggestion
The following examples show some of the ways in which you can configure logging in Quarkus:
```

== Centralized log management

>  Efficiently collect, store, and analyze log data from various components and instances of the application in a centralized location.

Reviewer 2: ```suggestion
Use a centralized location to efficiently collect, store, and analyze log data from various components and instances of the application.
```

==== Adding and reading MDC data

To add data to the MDC and extract it in your log output:

> . Use the `MDC` class to set the data
> . Customize the log format to use `%X\{mdc-key\}`

Reviewer 2: ```suggestion
. Use the `MDC` class to set the data.
. Customize the log format to use `%X\{mdc-key\}`.
```

Depending on the API you use, the MDC class is slightly different.
However, the APIs are very similar:

* Log4j 1 - `org.apache.log4j.MDC.put(key, value)`
* Log4j 2 - `org.apache.logging.log4j.ThreadContext.put(key, value)`

> * SLf4J - `org.slf4j.MDC.put(key, value)`

Reviewer: ```suggestion
* SLF4J - `org.slf4j.MDC.put(key, value)`
```

[[mdc-propagation]]
==== MDC propagation

In Quarkus, the MDC provider has a specific implementation for handling the reactive context, ensuring that MDC data is propagated during reactive and asynchronous processing.

As a result, you can still access the MDC data in various scenarios:

> * After asynchronous calls, like for example, when a REST client returns a Uni.

Reviewer 2: ```suggestion
* After asynchronous calls, for example, when a REST client returns a Uni.
```

* In code submitted to `org.eclipse.microprofile.context.ManagedExecutor`.
* In code executed with `vertx.executeBlocking()`.

> NOTE: If possible, MDC data is stored in a _duplicated context_, which is an isolated context for processing a single task (request).

Reviewer 2: This doesn't seem right. Should this be an instruction?

```suggestion
NOTE: Where possible, store MDC data in a _duplicated context_, which is an isolated context for processing a single task (request).
```
Or is this a clarification:

```suggestion
NOTE: In most instances, MDC data is stored in a _duplicated context_, which is an isolated context for processing a single task (request).
```
Or other?

Reviewer: This is _not_ an instruction. A duplicated context does not always exist, but when it is, Quarkus makes sure the MDC data are stored in it.

Reviewer 2: @Ladicek - I get it now. Thanks :+1: 
```suggestion
NOTE: If applicable, MDC data is stored in a _duplicated context_, which is an isolated context for processing a single task (request).
```

[[mdc-propagation]]
==== MDC propagation

In Quarkus, the MDC provider has a specific implementation for handling the reactive context, ensuring that MDC data is propagated during reactive and asynchronous processing.

As a result, you can still access the MDC data in various scenarios:

* After asynchronous calls, like for example, when a REST client returns a Uni.

> * When dealing with code submitted to the `ManagedExecutor` using `@Inject org.eclipse.microprofile.context.ManagedExecutor executor`.

Reviewer: ```suggestion
* In code submitted to `org.eclipse.microprofile.context.ManagedExecutor`.
```

* In code executed with `vertx.executeBlocking()`.

> NOTE: NOTE: MDC data is stored in a duplicated context, isolated for processing.

Reviewer: ```suggestion
NOTE: If possible, MDC data are stored in a _duplicated context_, which is an isolated context for processing a single task (request).
```

Reviewer: In my book, `data` is a plural, but what do I know :-)

Author: Thats another bad influence of playing Diablo 4 with all these typos and misconceptions :)
